PEPs 2, 4, 411: updates to how the stdlib is managed (#2651)
python/peps: pep-0002.txt

Title: Procedure for Adding New Modules
Version: $Revision$
Last-Modified: $Date$
Author: Martijn Faassen <faassen@infrae.com>
Status: Superseded
Author: Brett Cannon <brett@python.org>,
        Martijn Faassen <faassen@infrae.com>
Status: Active
Type: Process
Content-Type: text/x-rst
Created: 07-Jul-2001
Post-History: 07-Jul-2001, 09-Mar-2002

PEP Replacement
===============

This PEP has been superseded by the updated material in the Python

> Developer's Guide [1]_.

Reviewer: If this PEP was superseded, leave it as a historical document, and create a new one?

Reviewer 2: Indeed, very much agreed, see [my comment](https://github.com/python/peps/pull/2651#pullrequestreview-1019030218) to that effect.

> Due to the visibiity and importance of the standard library, it must

Reviewer: ```suggestion
Due to the visibility and importance of the standard library, it must
```

be maintained thoughtfully. Any code in the standard library must be
maintained by Python's development team, and so there is a perpetual
cost to each addition to the standard library. There is also a
cognitive cost for users in familiarizing themselves with what is in
the standard library. All of this requires taking a measured approach
towards what is allowed to be added to the standard library.

New functionality is commonly added to the library in the form of new
modules. This PEP will describe the procedure for the *addition* of
new modules.  :pep:`4` deals with procedures for deprecation of modules;
the *removal* of old and unused modules from the standard library.

Acceptance Procedure
====================

For top-level modules/packages, a PEP is required. The procedure for
writing a PEP is covered in :pep:`1`.

For submodules of a preexisting package in the standard library, it is
at the discretion of the general Python development team and its
members.

General guidance on what modules typically are accepted into the

> standard library, the overall process, etc. are covered in [1]_.

Reviewer 3: With the same change to footnote 1:
```suggestion
standard library, the overall process, etc. are covered in `the developer's
guide`_.
```

> guide [1]_.

Author: Because I wasn't trying to rewrite the entire PEP from scratch, and so changing its style was not a priority to me.

Maintenance Procedure
=====================

Anything accepted into the standard library is expected to be
primarily maintained there, within Python's development infrastructure.
While some members of the development team may choose to maintain a

> backport of a module outside of the standard library, it is up to them

Author: That's Jason's decision and it doesn't contradict this statement. The point that other core devs can view the stdlib copy as canonical and not concern themselves w/ code outside of the repo still holds.

to keep their external code in sync as appropriate.

References

> ==========

Reviewer 2: Sorry, forgot to include this in my initial review—

I can't quite reach it, but this can simply be removed after using a proper inline link instead of a footnote for the one reference within.

> to keep their external code in sync as appropriate and not force the
> Python development team to maintain the synchronicity.

Reviewer 3: Suggested to delete from "and not force" -- this seems just a repetition of the earlier idea, but in stronger language.
```suggestion
to keep their external code in sync as appropriate.
```
